composer/callbacks/mlperf.py

```diff
@@ -0,0 +1,377 @@
+                constants.SUBMISSION_BENCHMARK: self.benchmark,
+                constants.SUBMISSION_DIVISION: self.division,
+                constants.SUBMISSION_ORG: self.submitter,
+                constants.SUBMISSION_PLATFORM: self.system_name,
+                constants.SUBMISSION_STATUS: self.status,
+            })
+
+    def _create_submission_folders(self, root_folder: str, system_name: str, benchmark: str):
+        os.makedirs(root_folder, exist_ok=True)
+
+        results_folder = os.path.join(root_folder, 'results')
+        log_folder = os.path.join(root_folder, 'results', system_name)
+        benchmark_folder = os.path.join(log_folder, benchmark)
+        systems_folder = os.path.join(root_folder, 'systems')
+
+        os.makedirs(results_folder, exist_ok=True)
+        os.makedirs(log_folder, exist_ok=True)
+        os.makedirs(benchmark_folder, exist_ok=True)
+        os.makedirs(systems_folder, exist_ok=True)
+
+    def _log_dict(self, data: Dict[str, Any]):
+        for key, value in data.items():
+            self.mllogger.event(key=key, value=value)
+
+    def _get_accuracy(self, state: State):
+        if self.metric_name not in state.current_metrics[self.metric_label]:
+            raise ValueError('Accuracy must be a validation metric.')
+        return state.current_metrics[self.metric_label][self.metric_name]
+
+    def fit_start(self, state: State, logger: Logger) -> None:
+        if rank_zero():
+
+            if not isinstance(state.train_dataloader, DataLoader):
+                raise TypeError("train dataloader must be a torch dataloader")
+            if not isinstance(state.evaluators[0].dataloader.dataloader, DataLoader):
+                raise TypeError("eval dataset must be a torch dataloader.")
+            if state.train_dataloader.batch_size is None:
+                raise ValueError("Batch size is required to be set for dataloader.")
+            if len(state.evaluators) > 1:
+                raise ValueError("Only one evaluator is supported for the MLPerfCallback.")
+            if not isinstance(state.train_dataloader.dataset, Sized):
+                raise TypeError("Train dataset must have __len__ property")
+            if not isinstance(state.evaluators[0].dataloader.dataloader.dataset, Sized):
+                raise TypeError("The eval dataset must have __len__ property")
+
+            self._log_dict({
+                constants.SEED: state.seed,
+                constants.GLOBAL_BATCH_SIZE: state.train_dataloader.batch_size * dist.get_world_size(),
+                constants.GRADIENT_ACCUMULATION_STEPS: state.grad_accum,
+                constants.TRAIN_SAMPLES: len(state.train_dataloader.dataset),
+                constants.EVAL_SAMPLES: len(state.evaluators[0].dataloader.dataloader.dataset)
+            })
+
+            # optionally, upload the system description file
+            logger.file_artifact(LogLevel.FIT, self.system_desc_upload_name, self.systems_path)
```

Comment:
nit: this line could be moved up to line 177 in INIT, where the file is written. It will work down here too (just slightly less intuitive to follow)